app/main.py

```diff
@@ -1,3 +1,80 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: int | float
+    ) -> None:
+        self.km = km
+
+    @staticmethod
+    def other_type_check(other: Union[Distance, float, int]) -> bool:
+        if not isinstance(other, (Distance, float, int)):
+            raise TypeError("'other' is not 'Distance', 'float', 'int'!!!")
+        return isinstance(other, Distance)
```

Comment:
The method `other_type_check` is doing two things: it's checking the type of `other` and raising an exception if `other` is not of the expected type. It would be better to separate these responsibilities into two different methods. One method could be a private method `_check_type` that raises the exception if `other` is not of the expected type. The other method can be `is_distance` that returns True if `other` is an instance of `Distance`. This way, each method has a single responsibility which makes the code easier to understand and maintain.